apps/explorer/src/components/module/module-functions-interaction/index.tsx

```diff
@@ -32,32 +33,36 @@ export function ModuleFunctionsInteraction({
             .filter(([_, anFn]) => anFn.is_entry)
             .map(([fnName, details]) => ({ name: fnName, details }));
     }, [normalizedModule]);
-
-    if (error) {
-        return (
-            <Banner variant="error">
-                Error loading module <strong>{moduleName}</strong> details.
-            </Banner>
-        );
-    }
-
-    if (!isLoading && !executableFunctions.length) {
+    const isEmpty = !isLoading && !executableFunctions.length && !error;
+    if (isEmpty || error || isLoading) {
         return (
-            <Banner variant="message">No public entry functions found.</Banner>
+            <div className="h-full flex items-center justify-center">
+                {error ? (
+                    <Banner variant="error">
+                        Error loading module <strong>{moduleName}</strong>{' '}
+                        details.
+                    </Banner>
+                ) : isEmpty ? (
+                    <div className="text-body font-medium text-steel-dark">
+                        No public entry functions found.
+                    </div>
+                ) : (
+                    <LoadingSpinner text="Loading data" />
+                )}
+            </div>
         );
     }
-
-    return !isLoading && executableFunctions.length ? (
+    return (
         <div className="flex flex-col gap-3">
             {executableFunctions.map(({ name, details }) => (
                 <ModuleFunction
-                    key={name}
+                    key={[packageId, moduleName, name].join('::')}
```

Comment:
Names should be unique per package, so I'm not sure why you'd need to do this.

Reply:
I think there was a case where methods from other packages had the same names and that causes the collapse component to stay open (ModuleFunction is not being recreated)

Reply:
Ah, I see what you mean. I think it's probably better to add a key to the `ModuleFunctionsInteraction` component that is the module name to force the children to reset, but it's not a huge thing so I'm also okay with this.